api/apps/api/src/modules/scenarios-features/copy/copy-operation.service.ts

```diff
@@ -0,0 +1,36 @@
+import { Injectable } from '@nestjs/common';
+import { EntityManager } from 'typeorm';
+import { InjectEntityManager } from '@nestjs/typeorm';
+import { apiConnections } from '@marxan-api/ormconfig';
+import { CreateFeaturesCommand } from '../create-features.command';
+import { CopyQuery } from './copy-query.service';
+import { CopyDataProvider } from './copy-data-provider.service';
+
+@Injectable()
+export class CopyOperation {
+  constructor(
+    private readonly copyQuery: CopyQuery,
+    private readonly copyDataProvider: CopyDataProvider,
+    @InjectEntityManager(apiConnections.geoprocessingDB.name)
+    private readonly geoEntityManager: EntityManager,
+  ) {}
+
+  async copy(command: CreateFeaturesCommand): Promise<{ id: string }[]> {
+    const {
+      project,
+      protectedAreaFilterByIds,
+      planningAreaLocation,
+    } = await this.copyDataProvider.prepareData(command);
+    const { parameters, query } = this.copyQuery.prepareStatement(
+      command,
```

Comment:
does it need the command itself?

Reply:
the entire service could be in fact a command handler